ghost/members-api/lib/controllers/RouterController.js

```diff
@@ -483,19 +485,46 @@ module.exports = class RouterController {
                 // Validate requested newsletters
                 let {newsletters: requestedNewsletters} = req.body;
 
-                if (requestedNewsletters && requestedNewsletters.length > 0) {
-                    const newsletterIds = requestedNewsletters.map(newsletter => newsletter.id);
+                if (requestedNewsletters && requestedNewsletters.length > 0 && requestedNewsletters.every(newsletter => newsletter.name !== undefined)) {
+                    const newsletterNames = requestedNewsletters.map(newsletter => `'${newsletter.name}'`);
                     const newsletters = await this._newslettersService.browse({
-                        filter: `id:[${newsletterIds}]`,
-                        columns: ['id','status']
+                        filter: `name:[${newsletterNames}]`,
+                        columns: ['id','name','status']
                     });
 
-                    if (newsletters.length !== newsletterIds.length) {
-                        const validNewsletterIds = newsletters.map(newsletter => newsletter.id);
-                        const invalidNewsletterIds = newsletterIds.filter(id => !validNewsletterIds.includes(id));
+                    if (newsletters.length !== newsletterNames.length) {
+                        if (newsletters.length < newsletterNames.length) { //check for invalid newsletters
+                            const validNewsletters = newsletters.map(newsletter => `'${newsletter.name}'`);
+                            const invalidNewsletters = newsletterNames.filter(newsletter => !validNewsletters.includes(newsletter));
+
+                            throw new errors.BadRequestError({
+                                message: tpl(messages.invalidNewsletters, {newsletters: invalidNewsletters})
+                            });
+                        } else { //check for ambiguous newsletters
```

Comment:
Need to confirm if this is even possible. While testing I found that we can't create two newsletters with one name. Even if a newsletter is archived, it doesn't allow to create with the same name.

Reply:
aha thats great, we can drop the code needed for ambiguity then 🥳